qa/suites/orch/cephadm/with-work/tasks/rotate-ssh-keys.yaml

```diff
@@ -0,0 +1,54 @@
+tasks:
+- exec:
+    mon.a:
+      - |
+        set -ex
+        # generate a new priv/pub key pair
+        ssh-keygen -q -t rsa -N '' -C '' -f new-key
+        # store key in unused config-key store location so other hosts can access it
+        /home/ubuntu/cephtest/cephadm shell --mount new-key -- ceph config-key set mgr/cephadm/testing_priv_key -i /mnt/new-key
+        /home/ubuntu/cephtest/cephadm shell --mount new-key.pub -- ceph config-key set mgr/cephadm/testing_pub_key -i /mnt/new-key.pub
+- exec:
+    all-hosts:
+      - |
+        set -ex
+        # set up the pub key on each host
+        /home/ubuntu/cephtest/cephadm shell -- ceph config-key get mgr/cephadm/testing_pub_key > new-key.pub
+        cat new-key.pub | cat >> /root/.ssh/authorized_keys
+- exec:
+    mon.a:
+      - |
+        set -ex
+        NEW_SSH_PUB=$(cat new-key.pub)
+        NEW_SSH_PRIV=$(cat new-key)
+        # set the new keys to be used by cephadm for ssh
+        /home/ubuntu/cephtest/cephadm shell --mount new-key new-key.pub -- ceph cephadm set-priv-key -i /mnt/new-key
+        /home/ubuntu/cephtest/cephadm shell --mount new-key new-key.pub -- ceph cephadm set-pub-key -i /mnt/new-key.pub
+        # verify keys were properly set in config key store
+        CLUSTER_SSH_PRIV=$(/home/ubuntu/cephtest/cephadm shell -- ceph config-key get mgr/cephadm/ssh_identity_key)
+        CLUSTER_SSH_PUB=$(/home/ubuntu/cephtest/cephadm shell -- ceph config-key get mgr/cephadm/ssh_identity_pub)
+        if [ "$CLUSTER_SSH_PRIV" != "$NEW_SSH_PRIV" ]; then
+           printf "Cluster priv key\n\n$CLUSTER_SSH_PRIV\n\nis not equal to new ssh key\n\n$NEW_SSH_PRIV\n\n"
+           exit 1
+        else
+          echo "cluster private key successfully updated"
+        fi
+        if [ "$CLUSTER_SSH_PUB" != "$NEW_SSH_PUB" ]; then
+           printf "Cluster pub key\n\n$CLUSTER_SSH_PUB\n\nis not equal to new ssh key\n\n$NEW_SSH_PUB\n\n"
+           exit 1
+        else
+          echo "cluster pub key successfully updated"
+        fi
+        # verify all hosts are reachable with new ssh settings
+        HOSTNAMES=$(/home/ubuntu/cephtest/cephadm shell -- ceph orch host ls --format json | jq -r '.[] | .hostname')
+        for host in $HOSTNAMES; do
+          # cephadm puts most of the check-host output to stderr (even when no error)
```

Comment:
seems that something that probably we have to fix (not as part of this PR)

Reply:
I think (but didn't verify) this is done explicitly because some of the output is meant to be used by the mgr module when it calls check-host and then the part in stderr is meant just for human consumption

Reply:
yeah, I remember we had a couple of related BUGs (of using stderr for normal output) and I tought that all of them were solved. I'll open an issue to track this one as well.